charts/trino/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if .Values.networkPolicy.enabled }}
+{{- if eq "NodePort" .Values.service.type}}
+{{- fail "NetworkPolicy enforcement is not supported with NodePort Services, as traffic reaches the Pod through the node itself, bypassing Pod-level network controls." }}
+{{- end }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "trino.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "trino.labels" . | nindent 4 }}
+    app.kubernetes.io/component: network-policy
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "trino.selectorLabels" . | nindent 6 }}
+  policyTypes:
+    - Ingress
+  {{- with .Values.networkPolicy.egress }}
+    - Egress
+  egress:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  ingress:
+    - from:
```

Comment:
The complete values test is failing in the CI, where Prometheus can't talk to these pods. I think this default rule is too restrictive, and we should put it in an example, not the default.

From the docs:
> Network policies do not conflict; they are additive. If any policy or policies apply to a given pod for a given direction, the connections allowed in that direction from that pod is the union of what the applicable policies allow. Thus, order of evaluation does not affect the policy result.

Reply:
But weirdly, [this example](https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors) says pod selectors can be mixed with namespace selectors.

Can you try adding a label selector for Prometheus in the test?

Reply:
Locally the tests work without errors, and I am having a really hard time figuring out why they fail in the CI. I am splitting them up again just to diagnose this a bit better.

Reply:
Just a minor clarification
```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: release-name-trino
  namespace: trino
spec:
  podSelector: # this here selects the target pods of the policy inside the current namespace where the policy is created
    matchLabels:
      app.kubernetes.io/name: trino
      app.kubernetes.io/instance: release-name
  policyTypes:
    - Ingress
  ingress:
    - from: # this here selects the pods that can talk to the target, this configuration tells that only pods within the trino namespace, having these specific labels can talk to the target pods. Notice that podSelector and namespaceSelector are in the same list element. If I were to split them in two list elements, then the podSelector would allow all pods with those labels from all namespaces, while the namespaceSelector would allow all pods within the trino namespace.
        - podSelector:
            matchLabels:
              app.kubernetes.io/name: trino
              app.kubernetes.io/instance: release-name
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: trino
```

Reply:
How are you running tests locally? In CI, they generate a random namespace. This is probably not needed, neither in CI nor locally.

Reply:
I split the tests and the network_policy one still fails.
The only difference I see is that I am running kind 0.24.0 while the CI runs 0.22.0. I'm downgrading my kind version to 0.22.0

Reply:
Mystery solved: https://github.com/kubernetes-sigs/kind/releases/tag/v0.24.0
They've just added NetworkPolicy support in v0.24.0...

Reply:
Can you add `version` param for the `kind-action` in the GHA workflow? [The docs](https://github.com/helm/kind-action?tab=readme-ov-file#inputs) say it defaults to 0.21. It's better to have it explicitly set in our workflow anyway.

Reply:
I did just that! I am still polishing a couple of things so I converted the PR back to draft.